src/main/java/org/springframework/cloud/deployer/spi/cloudfoundry/AbstractCloudFoundryDeployer.java

```diff
@@ -91,11 +91,16 @@ String buildpack(AppDeploymentRequest request) {
 	 */
 	<T> Function<Mono<T>, Mono<T>> statusRetry(String id) {
 		long statusTimeout = this.deploymentProperties.getStatusTimeout();
-		long requestTimeout = Math.round(statusTimeout * 0.20); // wait 200ms with status timeout of 1000ms
-		long initialRetryDelay = Math.round(statusTimeout * 0.10); // wait 100ms with status timeout of 1000ms
-
-		return m -> m.timeout(Duration.ofMillis(requestTimeout))
-			.doOnError(e -> logger.error(String.format("Error getting status for %s within %sms, Retrying operation.", id, requestTimeout)))
+		long requestTimeout = Math.round(statusTimeout * 0.40); // wait 500ms with default status timeout of 2000ms
+		long initialRetryDelay = Math.round(statusTimeout * 0.10); // wait 200ms with status timeout of 2000ms
+
+		if (requestTimeout < 500L) {
```

Comment:
@markpollack should we really change something that was set by the user? I think we should WARN the user, but the idea of changing a predefined value seems too bullish to me